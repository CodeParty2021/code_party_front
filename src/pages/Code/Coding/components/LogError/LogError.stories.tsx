import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { LogError } from "./LogError";
import { LogPanel } from "../LogPanel/LogPanel";

export default {
  title: "components/LogError",
  component: LogError,
} as ComponentMeta<typeof LogError>;

const Template: ComponentStory<typeof LogError> = (args) => (
  <LogPanel state="log">
    <LogError {...args} />
  </LogPanel>
);

export const Default = Template.bind({});
Default.args = {};
