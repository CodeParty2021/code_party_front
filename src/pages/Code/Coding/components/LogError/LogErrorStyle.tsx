import styled, { css } from "styled-components";
import { RED_20, RED_50, RED_70 } from "styles/colors";

export type LogErrorStyleProps = {};

const defaultStyle = css`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 10px;

  color: ${RED_70};
  line-height: 160%;

  background: ${RED_20}50;
  border: 2px solid ${RED_50};
  border-radius: 8px;

  width: 100%;

  .log_error_label {
    font-size: 14px;
    font-weight: bold;
  }
`;

export const LogErrorStyle = styled.div<LogErrorStyleProps>`
  ${defaultStyle}
`;

LogErrorStyle.defaultProps = {};
