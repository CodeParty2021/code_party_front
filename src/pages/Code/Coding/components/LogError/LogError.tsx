import React from "react";
import { LogErrorStyle, LogErrorStyleProps } from "./LogErrorStyle";

type Props = LogErrorStyleProps;

export const LogError: React.FC<Props> = ({ children, ...styleProps }) => {
  return (
    <LogErrorStyle {...styleProps}>
      <span className="log_error_label">エラーがおきました！</span>
      <span>{children}</span>
    </LogErrorStyle>
  );
};
