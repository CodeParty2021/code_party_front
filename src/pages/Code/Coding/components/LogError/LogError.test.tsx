import React from "react";
import { LogError } from "./LogError";
import "jest-styled-components";
import { render } from "enzyme";

describe("<LogError />", () => {
  it("snapshot test", () => {
    const wrapper = render(<LogError>syntax error occurred!</LogError>);

    expect(wrapper).toMatchSnapshot();
  });
});
