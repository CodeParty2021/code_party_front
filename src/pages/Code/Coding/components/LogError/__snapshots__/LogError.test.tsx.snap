// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogError /> snapshot test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 10px;
  color: #C41C3B;
  line-height: 160%;
  background: #FFDBE050;
  border: 2px solid #F45D71;
  border-radius: 8px;
  width: 100%;
}

.c0 .log_error_label {
  font-size: 14px;
  font-weight: bold;
}

<div
  class="c0"
>
  <span
    class="log_error_label"
  >
    エラーがおきました！
  </span>
  <span>
    syntax error occurred!
  </span>
</div>
`;
